crates/model/src/lib.rs

```diff
@@ -4,6 +4,7 @@ pub mod app_data;
 pub mod auction;
 pub mod bytes_hex;
 pub mod interaction;
+pub mod nonzero_u256;
```

Comment:
I think the intention of the `model` crate was to hold the DTOs for the API service. With that in mind, I don't think `model` is an appropriate location for this.

This is more of a nit and unrelated to the merit of this PR - just bringing it up because I noticed it.